src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,45 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
 public class SalaryInfo {
+    private static final String DELIMITER = " ";
+    private static final DateTimeFormatter formatter = DateTimeFormatter.ofPattern("dd.MM.yyyy");
+    private static final String REPORT_HEAD = "Report for period ";
+    private static final String SEPARATOR = " - ";
+    private static final int POSITION_DATE = 0;
+    private static final int POSITION_NAME = 1;
+    private static final int POSITION_WORKING_HOUR = 2;
+    private static final int POSITION_INCOME_PER_HOUR = 3;
+
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        int[] resultSalary = new int[names.length];
+        StringBuilder report = new StringBuilder();
+        LocalDate dateFromLocalDate = LocalDate.parse(dateFrom, formatter);
+        LocalDate dateToLocalDate = LocalDate.parse(dateTo, formatter);
+        for (String record : data) {
```

Comment:
```suggestion
        for (String name: names) {
            for (String record : data) {
```
![image](https://user-images.githubusercontent.com/81781479/193588895-58c158b2-f817-4bd4-878f-d1adc552b6d2.png)

you can append result to stringbuilder inside for loop for names

Reply:
But if we do report for decade, we need get records (from database) each time when we select next user, I do not sure that it will be quicker than one time fetch date.

Reply:
We get records only once, and then go throw lines for all users.